File: зад1.java
import java.util.Scanner;

public class Main {
    // Функция проверки на простоту
    public static int findPrime(int originalNumber) {
        // Проверка простоты
        for (int i = 2; i <= Math.sqrt(originalNumber); i++) {
            if (originalNumber % i == 0) {
                return 0;
            }
        }
        return 1;
    }

    // Функция для поиска ближайшего простого числа
    public static int nearestPrimeNumber(int originalNumber) {
        int leftward = originalNumber;
        int right = originalNumber + 1;
        // Проверка слева, справа или само число простое
        while (true) {
            if (findPrime(leftward) == 1) {
                return leftward;
            }
             else if (findPrime(right) == 1) {
                return right;
            }
            leftward --;
            right++;
        }
    }
    //механизм, с помощью которого можно передавать в Java-приложение данные прямо при его запуске
    public static void main(String[] args) {
        //можно использовать созданный объект scanner для чтения данных.
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите число больше или равное 0: ");
        //получить целое число методом
        int number = scanner.nextInt();

        while (number < 0) {
            System.out.print("Не верно введено значение, введите заново: ");
            number = scanner.nextInt();
        }

        int primeNumber;
        if (number == 0 || number == 1) {
            primeNumber = 2;
        }
         else {
            primeNumber = nearestPrimeNumber(number);
        }

        // Вывод
        System.out.println("Ближайшее простое число находится на расстоянии " + Math.abs(number - primeNumber) + " чисел");
    }
}

File: зад1.kt
import kotlin.math.sqrt 
import java.util.Scanner 
fun findPrime(originalNumber: Int): Int { 
    for (i in 2..sqrt(originalNumber.toDouble()).toInt()) { 
        if (originalNumber % i == 0) { 
            return 0 
        } 
    } 
    return 1 
} 
 

fun nearestPrimeNumber(originalNumber: Int): Int { 
    var leftward = originalNumber 
    var right = originalNumber + 1 
    while (true) { 
        if (findPrime(leftward) == 1) { 
            return leftward 
        } else if (findPrime(right) == 1) { 
            return right 
        } 
        leftward-- 
        right++ 
    } 
} 
 
fun main() { 
   print("enter a value equal to or greater than 0: ") 
   val scanner = Scanner(System.`in`) 
   val number = scanner.nextInt()  
   scanner.close(); 
    val primeNumber: Int
    if (number < 0){
        print("Error") 
    } else{
        if (number == 0 || number == 1) { 
            primeNumber = 2 
        } else { 
            primeNumber = nearestPrimeNumber(number) 
        } 
        print(Math.abs(number - primeNumber))
    }
}
